cabal-install/src/Distribution/Client/SavedFlags.hs

```diff
@@ -51,6 +51,7 @@ readCommandFlags :: FilePath -> CommandUI flags -> IO flags
 readCommandFlags path command = do
   savedArgs <- fmap (fromMaybe []) (readSavedArgs path)
   case (commandParseArgs command True savedArgs) of
+    CommandDelegate -> undefined -- This is never evaluated
```

Comment:
If something is never intended to be evaluated, better it not be `undefined` but instead `error "CommandDelegate Flags evaluated, this should never occur"` or the like, so if things go wrong its easier to pin down the reason.